pkg/local/diff.go

```diff
@@ -202,20 +202,19 @@ func WalkS3(root string, callbackFunc func(p string, info fs.FileInfo, err error
 		}
 		if info.IsDir() {
 			// TODO: We don't return dir results for the listing, how will this effect directory markers, and can we even support directory markers?
-			// Save encountered directories in a sorted queue and compare them with the first appearance of an file in that level
-			heap.Push(&dirQueue, p+path.Separator) // add path separator to dir name and sort it later
+			// Save encountered directories in a min heap and compare them with the first appearance of a file in that level
+			heap.Push(&stringHeap, p+path.Separator) // add path separator to dir name and sort it later
 			return filepath.SkipDir
 		}
 
-		heap.Init(&dirQueue)
-		for dirQueue.Len() > 0 {
-			dir := heap.Pop(&dirQueue).(string)
+		for stringHeap.Len() > 0 {
+			dir := stringHeap.Peek().(string)
 			if p > dir {
+				heap.Pop(&stringHeap) // remove from queue
 				if err = WalkS3(dir, callbackFunc); err != nil {
 					return err
 				}
-			} else { // Put it back in its place
-				heap.Push(&dirQueue, dir)
+			} else {
 				break
 			}
```

Comment:
As always, I would prefer to reverse test, write line 212 as
```go
			if p <= dir {
				break
			}
```
and lose a level of indentation.

(This is of course not required)

Reply:
Done!